### PR TITLE
[Merged by Bors] - Feat : Add geometric distribution

### PR DESCRIPTION
**Pull request by JADekker**
Adds the geometric probability distribution

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by RemyDegenne** [comment]
The `p ∈ Ioc 0 1` hypotheses look strange. Why not `(hp_pos : 0 < p) (hp_le_one : p <= 1)`?
It looks like you have to turn the Ioc hypothesis back into simple inequalities several times to then use `positivity`, so there is a drawback to using Ioc. Also, to provide a `p ∈ Ioc 0 1` term to one of your lemmas, you would basically have to put both inequalities into a constructor, while you could just as easily have provided the inequalities separately.
What made you use Ioc?

**Comment by JADekker** [comment]
No particular reason, I just wanted to reduce the number of hypotheses, so Ioc uses one less hypothesis. however, I agree that it is quite clumsy. Shall I rewrite in terms of your hypotheses?

**Comment by RemyDegenne** [comment]
Yes, please use two separate hypotheses.

**Comment by JADekker** [comment]
I've made the necessary changes. Perhaps `measurable_geometricPmfReal` (line 65) is redundant now, as the proof is literally `by measurability`?

File: Mathlib/Probability/Distributions/Geometric.lean [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  exact this
+
+/-- The geometric pmf is positive for all natural numbers -/
+lemma geometricPmfReal_pos {p : ℝ} {n : ℕ} (hp_pos : 0 < p) (hp_lt_one : p < 1) :
+    0 < geometricPmfReal p n := by
+  rw [geometricPmfReal]
+  have : 0 < 1-p := sub_pos.mpr hp_lt_one
+  positivity
+
+lemma geometricPmfReal_nonneg {p : ℝ} {n : ℕ}  (hp_pos : 0 < p) (hp_le_one : p <= 1) :
```
Comment (RemyDegenne) [inline Mathlib/Probability/Distributions/Geometric.lean:50]:
```suggestion
lemma geometricPmfReal_nonneg {p : ℝ} {n : ℕ} (hp_pos : 0 < p) (hp_le_one : p <= 1) :
```

File: Mathlib/Probability/Distributions/Geometric.lean [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  `geometricPmf p = ENNReal.ofReal (geometricPmfReal p)`.
+* `geometricMeasure`: a geometric measure on `ℕ`, parametrized by its success probability `p`.
+-/
+
+open scoped ENNReal NNReal
+
+open MeasureTheory Real Set Filter Topology
+
+namespace ProbabilityTheory
+
```
Comment (RemyDegenne) [inline Mathlib/Probability/Distributions/Geometric.lean:27]: This is used everywhere below and could be deleted from all lemma statements:
```suggestion

variable {p : ℝ}

```

**Review by RemyDegenne — APPROVED** [review]
Thanks!
bors d+

**Comment by JADekker** [comment]
@RemyDegenne I've just made a new commit with your suggested changes AND name style changes consistent with the PR #9681 that you also accepted. I just wanted to check in that this was fine before merging?

**Comment by RemyDegenne** [comment]
It looks good! You can merge when CI succeeds.

**Pull request closed without merging**